target/linux/x86/image/grub-efi.cfg

```diff
@@ -6,9 +6,23 @@ set fallback="1"
 set timeout=@TIMEOUT@
 search.fs_label openwrt_boot root --hint (hd0)
 
+load_env -f /boot/grub/grubenv
+if test $attempt = "0+1+1"; then
+    set default=$fallback
+else
+    set attempt="$attempt+1"
+    save_env -f /boot/grub/grubenv attempt
+fi
+
 menuentry "@TITLE@" {
 	linux /boot/vmlinuz root=PARTLABEL=openwrt_rootfs rootwait @CMDLINE@ noinitrd
 }
+menuentry "@TITLE@ alternate partition" {
+	linux /boot/vmlinuz_alt root=PARTLABEL=openwrt_rootfs_alt rootwait @CMDLINE@ noinitrd
+}
 menuentry "@TITLE@ (failsafe)" {
 	linux /boot/vmlinuz failsafe=true root=PARTLABEL=openwrt_rootfs rootwait @CMDLINE@ noinitrd
 }
+menuentry "@TITLE@ alternate partition (failsafe)" {
+	linux /boot/vmlinuz_alt failsafe=true root=PARTLABEL=openwrt_rootfs_alt rootwait @CMDLINE@ noinitrd
+}
```

Comment:
I dislike this.  This seems like you're creating a primary and secondary, yet these should instead be equivalent.  I would prefer naming the kernels `/boot/vmlinuz-#`, using labels of `openwrt_rootfs-#` and `openwrt_overlay-#`.  This would allow for more than 2 installations (unlikely to be used often, but someone would want this).  Then switching to the _other_ pair would simply mean swapping the order.